payment/src/test/java/org/killbill/billing/payment/api/TestPaymentApi.java

```diff
@@ -1446,7 +1446,7 @@ public void testVerifyJanitorFromPendingDuringCompletionFlow() throws PaymentApi
             final Payment completedPayment = createPayment(TransactionType.AUTHORIZE, initialPayment.getId(), initialPayment.getExternalKey(), transactionExternalKey, authAmount, PaymentPluginStatus.PROCESSED);
             Assert.fail();
         } catch (final PaymentApiException e) {
-            Assert.assertEquals(e.getCode(), ErrorCode.PAYMENT_INVALID_OPERATION.getCode());
+            Assert.assertEquals(e.getCode(), ErrorCode.PAYMENT_ACTIVE_TRANSACTION_KEY_EXISTS.getCode());
```

Comment:
The authorize duplication will fail because of linked state machine has no entry from Auth_success to another authorize machine.
But I think it's better to indicate a "successful txn exists with this external key" error than the old check rely on the state machine.
This also answered the previous question about why the "Authorize" was picked up in the previous fixing. @daliwei @sbrossie 
But this fixing will change the behavior under this and also the use case below at line 1467, is there any concern with this modification? @sbrossie